src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1892,12 +1892,33 @@ static bool processSubscriptions
       //
       if (subCacheActive == false)
       {
-        BSONObj query  = BSON("_id" << OID(mapSubId));
-        BSONObj update = BSON("$set" <<
-                              BSON(CSUB_LASTNOTIFICATION << rightNow) <<
-                              "$inc" << BSON(CSUB_COUNT << (long long) 1));
+        BSONObj subOrig;
+        std::string err;
+        collectionFindOne(getSubscribeContextCollectionName(tenant), BSON("_id" << OID(mapSubId)), &subOrig, &err);
```

Comment:
You could keep the query definition above

```
BSONObj query  = BSON("_id" << OID(mapSubId));
```

and use

```
collectionFindOne(getSubscribeContextCollectionName(tenant), query, &subOrig, ...);
```

as query is the same form both DB operations in this block.

Reply:
fixed in b2b21a1